edc-extensions/bpn-validation/bpn-validation-core/src/main/java/org/eclipse/tractusx/edc/validation/businesspartner/functions/BusinessPartnerGroupFunction.java

```diff
@@ -120,20 +121,13 @@ public boolean evaluate(Operator operator, Object rightValue, Permission rule, P
             return false;
         }
 
-
         var bpn = participantAgent.getIdentity();
         var groups = store.resolveForBpn(bpn);
 
-        // BPN not found in database
-        if (groups.failed()) {
-            policyContext.reportProblem(groups.getFailureDetail());
-            return false;
-        }
-        var assignedGroups = groups.getContent();
+        var errorMessage = evaluateEmptyGroups(groups, operator, bpn);
```

Comment:
results should not be passed into methods. ever. they have a fluent API, so please use it. at this point though, the `evaluateEmptyGroups` method is not needed and its content can be inlined here.

Reply:
I simplified the validation and removed the method.